hooks/pre_validations/05-set_hostname.rb

```diff
@@ -0,0 +1,43 @@
+unless app_value(:hostname).nil?
+  hostname = app_value(:hostname)
+
+  command = "hostname \"#{hostname}\" 2>&1"
+  logger.debug(command)
+  unless kafo.noop?
+    result = `command`
+    if $?.exitstatus > 0
+      say "<%= color('Hostname setup failed: #{result}', :bad) %>"
+      kafo.class.exit(101)
+    end
+  end
+
+  filename = '/etc/sysconfig/network'
+  logger.debug("Writing hostname (#{hostname}) to #{filename}")
+  unless kafo.noop?
+    entry = "HOSTNAME=#{hostname}"
+    begin
+      text = File.read(filename) 
+      text.gsub!(/^HOSTNAME=.*?$/, entry)
+      text += "#{entry}\n" unless text =~ /^HOSTNAME=/
+      File.open(filename, "w") { |file| file.write(text) }
+    rescue => error
+      say "<%= color('Hostname setup failed: #{error}', :bad) %>"
+      kafo.class.exit(101)
+    end
+  end
+
+  # set /etc/hosts if needed
+  `hostname -f`
+  if $?.exitstatus > 0 
+    filename = '/etc/hosts'
+    logger.debug("hostname -f failed. Setting temporary hostname for loopback in #{filename}")
+    unless kafo.noop?
+      open(filename, 'a') do |f|
+        f << "# temporary hostname for loopback\n"
+        f << "127.0.0.1 #{hostname}\n"
+      end
+    end
+  end
```

Comment:
If I read this correctly this ends up with:

```
127.0.0.1 localhost
127.0.0.1 fqdn
```

I don't think this is correct. One IP must have one line record.

I'd suggest you to put /etc/hosts and /etc/sysconfig/network under puppet control if you really want to be the authority to setup hostname. But IMHO this should never be job of foreman/sat/fusor installer. There are some requirements that must be met and they inlucde proper hostname configuration.